delocate/tools.py

```diff
@@ -345,10 +574,68 @@ def set_install_id(filename: str, install_id: str, ad_hoc_sign: bool = True):
         replace_signature(filename, "-")
 
 
-RPATH_RE = re.compile(r"path (.*) \(offset \d+\)")
+RPATH_RE = re.compile(r"path (?P<rpath>.*) \(offset \d+\)")
+
 
+def _parse_otool_rpaths(stdout: str) -> Dict[str, List[str]]:
+    '''Return the rpaths of the library `filename`.
+
+    Parameters
+    ----------
+    stdout : str
+        The decoded stdout of an 'otool -l' command.
 
-def get_rpaths(filename):
+    Returns
+    -------
+    rpaths : dict of list of str
+        Where the key is the architecture and the value is the list of paths.
+        If the library has no rpaths then the values will be an empty list.
+
+    Examples
+    --------
+    >>> _parse_otool_rpaths("""
+    ... example.so:
+    ...     cmd LC_RPATH
+    ... cmdsize 0
+    ...    path /example/path (offset 0)
+    ...     cmd LC_RPATH
+    ... cmdsize 0
+    ...    path @loader_path (offset 0)
+    ... """)
+    {'': ['/example/path', '@loader_path']}
+    >>> _parse_otool_rpaths("""example.so:""")  # No rpaths.
+    {'': []}
+    >>> _parse_otool_rpaths("""
+    ... example.so (architecture x86_64):
+    ...     cmd LC_RPATH
+    ... cmdsize 0
+    ...    path path/x86_64 (offset 0)
+    ... example.so (architecture arm64):
+    ...     cmd LC_RPATH
+    ... cmdsize 0
+    ...    path path/arm64 (offset 0)
+    ... """)
+    {'x86_64': ['path/x86_64'], 'arm64': ['path/arm64']}
+    '''
+    rpaths: Dict[str, List[str]] = {}
+    for arch, lines in _parse_otool_listing(stdout).items():
+        rpaths[arch] = []
+        line_no = 0
+        while line_no < len(lines):
+            line = lines[line_no]
+            line_no += 1
+            if line != "cmd LC_RPATH":
+                continue
+            cmdsize, path = lines[line_no : line_no + 2]
+            assert cmdsize.startswith("cmdsize "), "Could not parse:\n{stdout}"
+            match_rpath = RPATH_RE.match(path)
+            assert match_rpath, "Could not parse:\n{stdout}"
+            rpaths[arch].append(match_rpath["rpath"])
+            line_no += 2
+    return rpaths
+
+
+def get_rpaths(filename: str) -> Tuple[str, ...]:
```

Comment:
Do we have tests somewhere for multiple architecture case?  Sorry if I missed it.

Reply:
Same here as `get_install_names`.